### ISSUE TITLE
Export report

### ISSUE THREAD

**Issue by dma** [subgraph/Vega#13]
I'm sure this is already a task but I'd like the ability to export and thus share my report. Exporting as a PDF or as xhtml would be nice.

**Comment by ghost** [comment]
Very useful feature. I need it very much. Thank you.

**Comment by jmoratilla** [comment]
+1 to add this reporting feature

**Comment by 264nm** [comment]
Just want to chime in on this and also say that exporting reports would be amazingly useful... Vega presents scan reports in the nicest manner out of all the GUI webapp vuln scanners and it seems a shame to waste it!

**Comment by infinitnet** [comment]
+1 on this one too, it would be awesome to have an option to export the scan results including problem descriptions.

**Comment by SistemasIntelygenz** [comment]
Pleaseee, we need it!

**Comment by dma** [comment]
Yes, we'll do this. We know it is the number one feature request. :)

**Comment by therealmaxmouse** [comment]
This software is really good but being able to show others the data without hours of manual work would make Vega amazing!

**Comment by ghost** [comment]
This

**Comment by alanfalima** [comment]
Is this being implemented? Any forecast? This should be really nice and will put Vega into the fight =P

**Comment by dma** [comment]
Hey, yes, it's coming, nearly done.

**Comment by niksmac** [comment]
awaiting to see this.

**Comment by stephandesouza** [comment]
Nearly when? :)

One of the best feature from Arachni is the HTML/PDF report, and lovely how Vega works.

**Comment by dma** [comment]
Coming next month. Thanks for the nice feedback @stephandesouza.

**Comment by andreyplotkin** [comment]
Is there any workaround meantime? (e.g temp files...)

**Comment by dma** [comment]
Just sharing the model.db file with another user who uses Vega to open it. I'm gonna try to get this badly wanted feature out quickly this month.

**Comment by pgarg0211** [comment]
I loved the tool. Is there any update on when reporting feature is getting released?

**Comment by mesmeric** [comment]
First time using the tool - great results - Exporting to XML, CSV, HTML would be very useful :)

**Comment by kartone** [comment]
Great tool, just waiting to have export feature ! :)

**Comment by spacetraveller** [comment]
That would be great!

**Comment by kangarolf** [comment]
Any news on the export tool..?

**Comment by dma** [comment]
Yes, HTML export is nearly done. Should be ready before the new year. Sorry about the way too long delay on this.

**Comment by GeorgeGkas-zz** [comment]
After one month I want to ask if it is done! I just download the Linux version and there is no export future yet.

**Comment by master3395** [comment]
Is it done, i'm done making screenshots, i'm not going to move/send the scan file to people.

**Comment by mesmeric** [comment]
Any news on the HTML report? We are well into the new year now :)

**Comment by spattisson** [comment]
I'd be happy to beta test it when the report export function becomes available ;-)

**Comment by arocha01** [comment]
Hi dma! Any news on the export tool? Best Regars!

**Comment by karthikeyan-kandasamy** [comment]
Hi dma, It would be great if the HTML export option released. Any news on this? :)

**Comment by dma** [comment]
Hey,

Just started working on it again today. Hopefully will get it done this
week.

Karthikeyan Kandasamy:

> Hi dma, It would be great if the HTML export option released. Any news on this? :)
> 
> ---
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/subgraph/Vega/issues/13#issuecomment-228267911

## 

David Mirza Ahmad
https://subgraph.com
78A1 CCFD 1C60 4BA7 5E1C C1F2 42D7 08C0 2520 8C7B

**Comment by waston4u** [comment]
hi dma
any news please?
cheers

**Comment by solo1317** [comment]
Me sumo a la espera de poder exportar reportes con vega, 
por el momento la unica forma es click derecho en cada vulnerabilidad y luego en ver codigo fuente
copio el codigo y lo pego en notepad al final pego el codigo de la siguiente vulnerabilidad hasta completarlas todas lugeo guardo el archivo como vegascan.html y ahi tengo mi reporte consolidado

**Comment by Insaida** [comment]
How long before this reporting feature comes out? Do you need help with writing it? I could help.

**Comment by kaitos90210** [comment]
I could help too!

**Comment by Insaida** [comment]
Hi @kaitos90210  We may have to do it on our own, seems like David isn't ready. I need this feature like last year. let me know if you're up for it.

**Comment by jamborloi** [comment]
Hi @dma  any news for the importing of the report?

**Comment by m-finn** [comment]
Any news on this? - just need the basic report to pass to the Developer. Its not very useful only me seeing the output, i cant do anything to fix them.

**Comment by SleepyLctl** [comment]
Naah still nothing? After several years?

**Comment by MRamonLeon** [comment]
I also need it

**Comment by espressobeanies** [comment]
If we gave it a bug bounty, would that give it the traction needed to be implemented?

**Comment by crmorris2** [comment]
I know this is an old thread, but instead of complaining about not getting features implemented in FREE software, why not look at other VERY easy solutions? Just right click on the console text > select all > ctrl+c > ctrl+v in notepad++ or word or any text editor. Job done!

**Comment by NasserMahmoud1** [comment]
Hi @dma,
It's been 2 years since your last comment,
I'm not sure if you are using the same calendar as we do, since "done this week" took 2 years :D

**Comment by mrahmatellah** [comment]
any update @dma

**Comment by anubi46** [comment]
without the export of the report it s useless, you will do some update or this application it s die?

**Comment by tanc7** [comment]
You guys, I found out (for Kali linux), that the results are stored in the /root/.vega directory.

For example, to extract all of the discovered emails from the email-finder module

cd /root/.vega
egrep -air mailto --color | awk -F 'mailto:' '{print $2}'

then you can pipe the command to a file for more manipulation.

**Comment by SeaWolf-666** [comment]
@dma,
You really have made an outstanding tool, but the lack of an export report tool is killing me.(jk lol). Regardless, thank you for the work you have done on this.

Anyways, for anyone else looking for this, I have another tip. To add to @tanc7 last reply, you can also use "strings /root/.vega/workspaces/00/model.db >> index.html" to make an HTML file with the pages Vega found that may be an issue. For example, if a directory has directory listing enabled, you will see a copy of that directory in the output of the index.html the above command will create. This may or may not be helpful for some.

It's not a full report, but it is a little more info you can use.

**Comment by pomil-1969** [comment]
> You guys, I found out (for Kali linux), that the results are stored in the /root/.vega directory.
> 
> For example, to extract all of the discovered emails from the email-finder module
> 
> cd /root/.vega
> egrep -air mailto --color | awk -F 'mailto:' '{print $2}'
> 
> then you can pipe the command to a file for more manipulation.

Hello tanc7,

because I am new with this, 
how do I pipe the command to a file?
Could you please tell me step-by-step?
I need this urgently to export the results to a report to show to my bosses.

**Comment by Insaida** [comment]
Hey pomil-1969

> how do I pipe the command to a file?

You can pipe the output of one command into another command using the `>>` characters.

for example, if i wanted the output of `apt install vega` to be written to a file, I just pipe it to a file name by doing `ls -lha >> file.txt`. That's how you pipe the contents of a command to a file.

**Comment by uniuuu** [comment]
+1 to make report feature.



> Just sharing the model.db file with another user who uses Vega to open it. I'm gonna try to get this badly wanted feature out quickly this month.

Hi @dma 
Maybe people can contribute to it. 
Could you please share how's model.db made? Can try to think of SQL query to export.

**Comment by anneborcherding** [comment]
It is possible to export the scan results as HTML if you use the development branch of Vega. Note that the exported html file does not care about the given file path but will be saved in /tmp/test.html no matter what. If you would like to export the scan results automatically, have a look at [my fork](https://github.com/anneborcherding/Vega) which provides a python interface for Vega.

**Comment by tanc7** [comment]
Whoa. Time sure has passed. Sorry I missed all of your messages.
